tests/helpers/core-manager.js

```diff
@@ -40,3 +42,37 @@ export function replicate(cm1, cm2) {
     ])
   }
 }
+
+export async function waitForCores (coreManager, keys) {
+  const allKeys = getAllKeys(coreManager)
+  if (hasKeys(keys, allKeys)) return
+  return new Promise(res => {
+    coreManager.on('add-core', async function onAddCore ({ key, core }) {
+      await core.ready()
```

Comment:
Added `await core.ready()` here for convenience. Did a quick check to make sure that still works for the core-manager tests and it seems fine.